### ISSUE TITLE
Expo has not been este against Node.js v18.12.1

### ISSUE THREAD

**Issue by AlejandroJBArellano** [expo/expo-cli#4619]
### Summary

I ran: ``npm start``

And then pressed ``w`` to see my app in the web browser.

But this error stopped me: 

```
node:internal/crypto/hash:71
  this[kHandle] = new _Hash(algorithm, xofLen);
                  ^

Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:71:19)
    at Object.createHash (node:crypto:133:10)
    at module.exports (C:\Users\Peridot\Escritorio\proyectos\DoneWithIt\node_modules\webpack\lib\util\createHash.js:135:53)
    at NormalModule._initBuildHash (C:\Users\Peridot\Escritorio\proyectos\DoneWithIt\node_modules\webpack\lib\NormalModule.js:417:16)
    at handleParseError (C:\Users\Peridot\Escritorio\proyectos\DoneWithIt\node_modules\webpack\lib\NormalModule.js:471:10)
    at C:\Users\Peridot\Escritorio\proyectos\DoneWithIt\node_modules\webpack\lib\NormalModule.js:503:5
    at C:\Users\Peridot\Escritorio\proyectos\DoneWithIt\node_modules\webpack\lib\NormalModule.js:358:12
    at C:\Users\Peridot\Escritorio\proyectos\DoneWithIt\node_modules\loader-runner\lib\LoaderRunner.js:373:3
    at iterateNormalLoaders (C:\Users\Peridot\Escritorio\proyectos\DoneWithIt\node_modules\loader-runner\lib\LoaderRunner.js:214:10)
    at Array.<anonymous> (C:\Users\Peridot\Escritorio\proyectos\DoneWithIt\node_modules\loader-runner\lib\LoaderRunner.js:205:4)
    at Storage.finished (C:\Users\Peridot\Escritorio\proyectos\DoneWithIt\node_modules\enhanced-resolve\lib\CachedInputFileSystem.js:55:16)
    at C:\Users\Peridot\Escritorio\proyectos\DoneWithIt\node_modules\enhanced-resolve\lib\CachedInputFileSystem.js:91:9
    at C:\Users\Peridot\Escritorio\proyectos\DoneWithIt\node_modules\graceful-fs\graceful-fs.js:123:16
    at FSReqCallback.readFileAfterClose [as oncomplete] (node:internal/fs/read_file_context:68:3) {
  opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
}

Node.js v18.12.1
```

### Environment

```
  expo-env-info 1.0.5 environment info:
    System:
      OS: Windows 10 10.0.19045
    Binaries:
      Node: 18.12.1 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
      npm: 8.5.5 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version     2020.3.0.0 AI-203.7717.56.2031.7784292
    npmPackages:
      @expo/webpack-config: ^0.17.2 => 0.17.3
      expo: ~47.0.9 => 47.0.11
      react: 18.1.0 => 18.1.0
      react-dom: 18.1.0 => 18.1.0
      react-native: 0.70.5 => 0.70.5
      react-native-web: ~0.18.9 => 0.18.10
    Expo Workflow: managed

```

### Please specify your device/emulator/simulator platform, model and version

Web. Node.js v18.12.1

### Error output

```
Starting Webpack on port 19006 in development mode.
‼ ｢wds｣: transportMode is an experimental option, meaning its usage could potentially change without warning
  expo:start:server:developmentSession Development session will not ping because the user is not authenticated and there are no devices. +5s
  expo:start:server:urlCreator URL: ... +5s
  expo:start:server:devServer Redirect page is disabled +2s
"QR Stuff"

› Metro waiting on ...
› Scan the QR code above with Expo Go (Android) or the Camera app (iOS)

› Web is waiting on http://localhost:19006

› Press a │ open Android
› Press w │ open web

› Press j │ open debugger
› Press r │ reload app
› Press m │ toggle menu

› Press ? │ show all commands

› Open in the web browser...
› Press ? │ show all commands
node:internal/crypto/hash:71
  this[kHandle] = new _Hash(algorithm, xofLen);
                  ^

Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:71:19)
    at Object.createHash (node:crypto:133:10)
    at module.exports (C:\Users\Peridot\Escritorio\proyectos\DoneWithIt\node_modules\webpack\lib\util\createHash.js:135:53)
    at NormalModule._initBuildHash (C:\Users\Peridot\Escritorio\proyectos\DoneWithIt\node_modules\webpack\lib\NormalModule.js:417:16)
    at handleParseError (C:\Users\Peridot\Escritorio\proyectos\DoneWithIt\node_modules\webpack\lib\NormalModule.js:471:10)
    at C:\Users\Peridot\Escritorio\proyectos\DoneWithIt\node_modules\webpack\lib\NormalModule.js:503:5
    at C:\Users\Peridot\Escritorio\proyectos\DoneWithIt\node_modules\webpack\lib\NormalModule.js:358:12
    at C:\Users\Peridot\Escritorio\proyectos\DoneWithIt\node_modules\loader-runner\lib\LoaderRunner.js:373:3
    at iterateNormalLoaders (C:\Users\Peridot\Escritorio\proyectos\DoneWithIt\node_modules\loader-runner\lib\LoaderRunner.js:214:10)
    at Array.<anonymous> (C:\Users\Peridot\Escritorio\proyectos\DoneWithIt\node_modules\loader-runner\lib\LoaderRunner.js:205:4)
    at Storage.finished (C:\Users\Peridot\Escritorio\proyectos\DoneWithIt\node_modules\enhanced-resolve\lib\CachedInputFileSystem.js:55:16)
    at C:\Users\Peridot\Escritorio\proyectos\DoneWithIt\node_modules\enhanced-resolve\lib\CachedInputFileSystem.js:91:9
    at C:\Users\Peridot\Escritorio\proyectos\DoneWithIt\node_modules\graceful-fs\graceful-fs.js:123:16
    at FSReqCallback.readFileAfterClose [as oncomplete] (node:internal/fs/read_file_context:68:3) {
  opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
}

Node.js v18.12.1

```

### Reproducible demo or steps to reproduce from a blank project

1. Use the blanck template with typescript.
2. Run ```npx expo install @expo/webpack-config@^0.17.2```
3. Run ```npm start``` and select the browser option ``w``

**Comment by RyelBanfield** [comment]
Is the only fix here to use node 16?

**Comment by RyelBanfield** [comment]
For anyone looking for a temporary fix.

https://github.com/expo/expo-cli/issues/4575#issuecomment-1293202536

**Comment by EvanBacon** [comment]
Closing in favor of #4575

**Comment by diegoolipa** [comment]
I solved it with this method:

I had installed: Node.js v18.12.1

I solved the problem by installing Node.js v14.17.3

**Comment by lawyinking** [comment]
> export NODE_OPTIONS=--openssl-legacy-provider

this helps, thanks a lot!!!!

**Comment by Nditah** [comment]
https://stackoverflow.com/questions/72866798/node-openssl-legacy-provider-is-not-allowed-in-node-options

QUICK FIX

You can try downgrade the version. But as a quick fix you can do the below options.

On Unix-like (Linux, macOS, Git bash, etc.):

    export NODE_OPTIONS=--openssl-legacy-provider

On Windows command prompt:

    set NODE_OPTIONS=--openssl-legacy-provider

On PowerShell:

    $env:NODE_OPTIONS = "--openssl-legacy-provider"

[Read more](https://stackoverflow.com/questions/69692842/error-message-error0308010cdigital-envelope-routinesunsupported)

**Comment by viviramji** [comment]
It worked with your reply @Nditah, thank u very much! :D
